solutions/fully-configurable/main.tf

```diff
@@ -32,3 +42,139 @@ module "scc_wp" {
+  source                      = "terraform-ibm-modules/trusted-profile/ibm"
+  version                     = "2.1.1"
+  trusted_profile_name        = local.config_service_trusted_profile_name
+  trusted_profile_description = "Trusted Profile for App Config"
+
+  trusted_profile_identity = {
+    identifier    = var.app_config_crn
+    identity_type = "crn"
+  }
+
+  trusted_profile_policies = [
+    {
+      roles = ["Viewer", "Service Configuration Reader"]
+      resources = [{
+        service = "All Account Management services"
+      }]
+      description = "All Account Management services"
+    },
+    {
+      roles = ["Viewer", "Service Configuration Reader"]
+      resources = [{
+        service = "All Identity and Access enabled services"
+      }]
+      description = "All Identity and Access enabled services"
+    },
+  ]
+}
+
+##############################################################
+# CRN Parser
+##############################################################
+
+module "crn_parser" {
+  count   = var.cspm_enabled ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.app_config_crn
+}
+
+################################################################
+# Config Service Instance
+################################################################
+
+resource "ibm_config_aggregator_settings" "config_aggregator_settings_instance" {
+  count       = var.cspm_enabled ? 1 : 0
+  instance_id = module.crn_parser[0].service_instance
+  region      = var.region
+
+  resource_collection_regions = ["all"]
+  resource_collection_enabled = true
+  trusted_profile_id          = module.trusted_profile_app_config[0].profile_id
+}
+
+################################################################
+# IAM Auth Token
+################################################################
+
+data "ibm_iam_auth_token" "auth_token" {}
+
+################################################################
+# Enable CSPM for SCC Workload Protection instance
+################################################################
+
+# CSPM can only be enabled after the trusted profile exists,
+# but profile can only exist after instance has been created
+# hence we cannot directly enable CSPM in the instance creation
+# and need to use a separate resource to enable it
+resource "restapi_object" "enable_cspm" {
+  count = var.cspm_enabled ? 1 : 0
+
+  path = "/v2/resource_instances/${module.scc_wp.guid}"
+
+  data = jsonencode({
+    parameters = {
+      enable_cspm = true
+      target_accounts = [
+        {
+          account_id         = local.account_id
+          config_crn         = var.app_config_crn
+          trusted_profile_id = module.trusted_profile_scc_wp[0].profile_id
+        }
+      ]
+    }
+  })
+  create_method = "PATCH" # Specify the HTTP method for updates
+
+  depends_on = [
+    module.scc_wp,
+    module.trusted_profile_scc_wp
```

Comment:
you don't need either of these, there is already implicit dependencies on line 169 and 160